src/material-experimental/mdc-slider/slider.ts

```diff
@@ -6,13 +6,163 @@
  * found in the LICENSE file at https://angular.io/license
  */
 
-import {ChangeDetectionStrategy, Component, ViewEncapsulation} from '@angular/core';
+import {
+  ChangeDetectionStrategy,
+  Component,
+  ViewEncapsulation,
+} from '@angular/core';
+import {SpecificEventListener, EventType} from '@material/base';
+import {
+  MDCSliderAdapter,
+  MDCSliderFoundation,
+  Thumb,
+  TickMark,
+} from '@material/slider';
 
+/**
+ * Allows users to select from a range of values by moving the slider thumb. It is similar in
+ * behavior to the native `<input type="range">` element.
+ */
 @Component({
   selector: 'mat-slider',
   templateUrl: 'slider.html',
   styleUrls: ['slider.css'],
   changeDetection: ChangeDetectionStrategy.OnPush,
   encapsulation: ViewEncapsulation.None,
 })
-export class MatSlider {}
+export class MatSlider {
+
+  /** Instance of the MDC slider foundation for this slider. */
+
+  // tslint:disable-next-line:no-unused-variable
+  private _foundation = new MDCSliderFoundation(new SliderAdapter());
+}
+
+class SliderAdapter implements MDCSliderAdapter {
+  constructor() {}
+  hasClass(className: string): boolean {
+    throw new Error('Method not implemented.');
```

Comment:
Not important now since this is just WIP, but FYI you can omit the `new` and do just `throw Error(...)`

(omitting `new` isn't allowed by Google's internal typescript style guide, but we do it here on github)